# Curso de Estatística para Análise de Dados com Python

## Tratamento dos dados referentes ao Enem 2019 no estado de São Paulo.
 - O objetivo do curso é analisar os dados referentes a esse arquivo. O arquivo pode ser encontrado no site do INEP.
 - De todo o Brasil: 
http://inep.gov.br/microdados

In [1]:
import numpy as np
import pandas as pd

In [3]:
dados = pd.read_csv('microdados_enem_2019_sp.csv', sep=';', encoding='iso-8859-1')

### Extração e Limpeza dos Dados

In [4]:
# Visualização
dados.head()

NU_INSCRICAO  CO_MUNICIPIO_RESIDENCIA NO_MUNICIPIO_RESIDENCIA  \
0     1,90E+11                  3546801            Santa Isabel   
1     1,90E+11                  3538204             Pinhalzinho   
2     1,90E+11                  3550308               São Paulo   
3     1,90E+11                  3548708   São Bernardo do Campo   
4     1,90E+11                  3549409    São Joaquim da Barra   

   CO_UF_RESIDENCIA SG_UF_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0                35               SP        22       M                1   
1                35               SP        19       M                1   
2                35               SP        17       F                1   
3                35               SP        19       M                1   
4                35               SP        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  ...  NU_NOTA_MT TP_LINGUA  \
0            3                 1  ...      6002.0         0   
1            3                 1  ...         NaN         1   
2            3                 1  ...      7318.0         0   
3            3                 1  ...      3932.0         1   
4            1                 1  ...         NaN         1   

   TP_STATUS_REDACAO NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  \
0                1.0         160.0          200.0          180.0   
1                NaN           NaN            NaN            NaN   
2                1.0         160.0          120.0          200.0   
3                1.0          80.0          160.0          120.0   
4                NaN           NaN            NaN            NaN   

   NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          200.0          200.0            940.0     B  
1            NaN            NaN              NaN     A  
2          200.0          200.0            880.0     A  
3          100.0          100.0            560.0     B  
4            NaN            NaN              NaN     B  

[5 rows x 44 columns]

In [ ]:
# linhas e colunas (Linhas serão a quantidade de inscritos)
dados.shape

(813772, 44)

In [6]:
# análise dos tipos de atributos
dados.dtypes

NU_INSCRICAO                object
CO_MUNICIPIO_RESIDENCIA      int64
NO_MUNICIPIO_RESIDENCIA     object
CO_UF_RESIDENCIA             int64
SG_UF_RESIDENCIA            object
NU_IDADE                     int64
TP_SEXO                     object
TP_ESTADO_CIVIL              int64
TP_COR_RACA                  int64
TP_NACIONALIDADE             int64
CO_MUNICIPIO_NASCIMENTO    float64
NO_MUNICIPIO_NASCIMENTO     object
CO_UF_NASCIMENTO           float64
SG_UF_NASCIMENTO            object
TP_ST_CONCLUSAO              int64
TP_ANO_CONCLUIU              int64
TP_ESCOLA                    int64
TP_ENSINO                  float64
IN_TREINEIRO                 int64
CO_ESCOLA                  float64
CO_MUNICIPIO_ESC           float64
NO_MUNICIPIO_ESC            object
CO_UF_ESC                  float64
SG_UF_ESC                   object
TP_DEPENDENCIA_ADM_ESC     float64
TP_LOCALIZACAO_ESC         float64
TP_SIT_FUNC_ESC            float64
TP_PRESENCA_CN               int64
TP_PRESENCA_CH      

In [7]:
# Excluindo uma coluna (colunas que não são interessantes para análise)
dados1 = dados.drop(columns=['CO_MUNICIPIO_RESIDENCIA'])

In [8]:
# Conferindo se a exclusão deu certo
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  CO_UF_RESIDENCIA SG_UF_RESIDENCIA  \
0     1,90E+11            Santa Isabel                35               SP   
1     1,90E+11             Pinhalzinho                35               SP   
2     1,90E+11               São Paulo                35               SP   
3     1,90E+11   São Bernardo do Campo                35               SP   
4     1,90E+11    São Joaquim da Barra                35               SP   

   NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  TP_COR_RACA  TP_NACIONALIDADE  \
0        22       M                1            3                 1   
1        19       M                1            3                 1   
2        17       F                1            3                 1   
3        19       M                1            3                 1   
4        37       M                2            1                 1   

   CO_MUNICIPIO_NASCIMENTO  ... NU_NOTA_MT  TP_LINGUA TP_STATUS_REDACAO  \
0                1506500.0  ...     6002.0          0               1.0   
1                1503804.0  ...        NaN          1               NaN   
2                1501402.0  ...     7318.0          0               1.0   
3                3548708.0  ...     3932.0          1               1.0   
4                3549409.0  ...        NaN          1               NaN   

   NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  \
0          160.0          200.0          180.0          200.0          200.0   
1            NaN            NaN            NaN            NaN            NaN   
2          160.0          120.0          200.0          200.0          200.0   
3           80.0          160.0          120.0          100.0          100.0   
4            NaN            NaN            NaN            NaN            NaN   

   NU_NOTA_REDACAO  Q025  
0            940.0     B  
1              NaN     A  
2            880.0     A  
3            560.0     B  
4              NaN     B  

[5 rows x 43 columns]

In [9]:
# Exclusão de outras colunas (presenã do , no final ao cortar mais de uma coluna)
dados1 = dados1.drop(columns=['CO_UF_RESIDENCIA','SG_UF_RESIDENCIA','CO_MUNICIPIO_NASCIMENTO'], )

In [10]:
dados1.head(15)

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0      1,90E+11            Santa Isabel        22       M                1   
1      1,90E+11             Pinhalzinho        19       M                1   
2      1,90E+11               São Paulo        17       F                1   
3      1,90E+11   São Bernardo do Campo        19       M                1   
4      1,90E+11    São Joaquim da Barra        37       M                2   
5      1,90E+11              Embu-Guaçu        17       F                1   
6      1,90E+11     São José dos Campos        28       F                1   
7      1,90E+11     São José dos Campos        25       F                2   
8      1,90E+11               Guarulhos        23       F                1   
9      1,90E+11               São Paulo        19       F                1   
10     1,90E+11   São José do Rio Preto        18       F                1   
11     1,90E+11                  Sumaré        22       F                1   
12     1,90E+11                  Franca        32       M                1   
13     1,90E+11               São Paulo        19       F                1   
14     1,90E+11                   Bauru        18       M                1   

    TP_COR_RACA  TP_NACIONALIDADE NO_MUNICIPIO_NASCIMENTO  CO_UF_NASCIMENTO  \
0             3                 1    Santa Izabel do Pará              15.0   
1             3                 1                 Jacundá              15.0   
2             3                 1                   Belém              15.0   
3             3                 1   São Bernardo do Campo              35.0   
4             1                 1    São Joaquim da Barra              35.0   
5             3                 1             Itacoatiara              13.0   
6             1                 1                  Manaus              13.0   
7             1                 1     São José dos Campos              35.0   
8             1                 1               São Paulo              35.0   
9             3                 1                   Belém              15.0   
10            1                 1                   Belém              15.0   
11            4                 1              Ananindeua              15.0   
12            3                 1              Pratápolis              31.0   
13            1                 1                   Belém              15.0   
14            3                 1                    Jaru              11.0   

   SG_UF_NASCIMENTO  ...  NU_NOTA_MT  TP_LINGUA  TP_STATUS_REDACAO  \
0                PA  ...      6002.0          0                1.0   
1                PA  ...         NaN          1                NaN   
2                PA  ...      7318.0          0                1.0   
3                SP  ...      3932.0          1                1.0   
4                SP  ...         NaN          1                NaN   
5                AM  ...      4192.0          0                1.0   
6                AM  ...         NaN          0                NaN   
7                SP  ...         NaN          1                NaN   
8                SP  ...      7961.0          0                1.0   
9                PA  ...         NaN          0                NaN   
10               PA  ...      6128.0          1                1.0   
11               PA  ...         NaN          1                NaN   
12               MG  ...      5841.0          1                1.0   
13               PA  ...         NaN          0                NaN   
14               RO  ...         NaN          1                NaN   

    NU_NOTA_COMP1  NU_NOTA_COMP2  NU_NOTA_COMP3  NU_NOTA_COMP4 NU_NOTA_COMP5  \
0           160.0          200.0          180.0          200.0         200.0   
1             NaN            NaN            NaN            NaN           NaN   
2           160.0          120.0          200.0          200.0         200.0   
3            80.0          160.0          120.0 

In [11]:
# Excluindo outras colunas que não serão utilizadas no estudo
dados1 = dados1.drop(columns=['NO_MUNICIPIO_NASCIMENTO','CO_UF_NASCIMENTO','SG_UF_NASCIMENTO',
                              'TP_ANO_CONCLUIU','TP_ENSINO','CO_MUNICIPIO_ESC','CO_UF_ESC','SG_UF_ESC'],)

In [12]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...      6002.0         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...      7318.0         0                1.0          160.0          120.0   
3  ...      3932.0         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  

[5 rows x 32 columns]

In [13]:
dados1 = dados1.drop(columns=['TP_DEPENDENCIA_ADM_ESC','TP_LOCALIZACAO_ESC','TP_SIT_FUNC_ESC'],)

In [14]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...      6002.0         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...      7318.0         0                1.0          160.0          120.0   
3  ...      3932.0         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  

[5 rows x 29 columns]

In [15]:
dados1.shape

(813772, 29)

In [17]:
# Correção do erro nas notas das disciplinas
# Como corrigir? 
# Dividir todas as colunas dessas disciplinas por 10

dados1.loc[:, 'NU_NOTA_CN'] /=10

In [19]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...      6002.0         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...      7318.0         0                1.0          160.0          120.0   
3  ...      3932.0         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  

[5 rows x 29 columns]

In [20]:
# Corrigindo de todas
dados1.loc[:, 'NU_NOTA_CH'] /=10
dados1.loc[:, 'NU_NOTA_LC'] /=10
dados1.loc[:, 'NU_NOTA_MT'] /=10

In [21]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...       600.2         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...       731.8         0                1.0          160.0          120.0   
3  ...       393.2         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NU_NOTA_REDACAO  Q025  
0          180.0          200.0          200.0            940.0     B  
1            NaN            NaN            NaN              NaN     A  
2          200.0          200.0          200.0            880.0     A  
3          120.0          100.0          100.0            560.0     B  
4            NaN            NaN            NaN              NaN     B  

[5 rows x 29 columns]

In [22]:
# Renomeando os nomes das colunas
dados1 = dados1.rename(columns={'NU_NOTA_REDACAO': 'NOTA_REDACAO'})

In [23]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  NU_IDADE TP_SEXO  TP_ESTADO_CIVIL  \
0     1,90E+11            Santa Isabel        22       M                1   
1     1,90E+11             Pinhalzinho        19       M                1   
2     1,90E+11               São Paulo        17       F                1   
3     1,90E+11   São Bernardo do Campo        19       M                1   
4     1,90E+11    São Joaquim da Barra        37       M                2   

   TP_COR_RACA  TP_NACIONALIDADE  TP_ST_CONCLUSAO  TP_ESCOLA  IN_TREINEIRO  \
0            3                 1                1          1             0   
1            3                 1                1          1             0   
2            3                 1                2          3             0   
3            3                 1                2          2             0   
4            1                 1                1          1             0   

   ...  NU_NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  NU_NOTA_COMP1  NU_NOTA_COMP2  \
0  ...       600.2         0                1.0          160.0          200.0   
1  ...         NaN         1                NaN            NaN            NaN   
2  ...       731.8         0                1.0          160.0          120.0   
3  ...       393.2         1                1.0           80.0          160.0   
4  ...         NaN         1                NaN            NaN            NaN   

   NU_NOTA_COMP3  NU_NOTA_COMP4  NU_NOTA_COMP5  NOTA_REDACAO  Q025  
0          180.0          200.0          200.0         940.0     B  
1            NaN            NaN            NaN           NaN     A  
2          200.0          200.0          200.0         880.0     A  
3          120.0          100.0          100.0         560.0     B  
4            NaN            NaN            NaN           NaN     B  

[5 rows x 29 columns]

In [24]:
# Renomeando nome das colunas de notas das disciplinas
dados1 = dados1.rename(columns={'NU_NOTA_CN': 'NOTA_CN','NU_NOTA_CH': 'NOTA_CH',
                                'NU_NOTA_LC': 'NOTA_LC','NU_NOTA_MT': 'NOTA_MT'})

In [25]:
# Renomeando nomes das colunas de notas das competencias
dados1 = dados1.rename(columns={'NU_NOTA_COMP1': 'COMP1','NU_NOTA_COMP2': 'COMP2',
                                'NU_NOTA_COMP3': 'COMP3','NU_NOTA_COMP4': 'COMP4',
                                'NU_NOTA_COMP5':'COMP5'})

In [26]:
# Renomeando nome das colunas de dados pessoais
dados1 = dados1.rename(columns={'NU_IDADE': 'IDADE','TP_SEXO': 'SEXO',
                                'TP_COR_RACA': 'RACA','Q025': 'INTERNET',
                                'TP_ESCOLA':'ESCOLA'})

In [29]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO  ESCOLA  IN_TREINEIRO  ...  NOTA_MT  \
0                 1                1       1             0  ...    600.2   
1                 1                1       1             0  ...      NaN   
2                 1                2       3             0  ...    731.8   
3                 1                2       2             0  ...    393.2   
4                 1                1       1             0  ...      NaN   

  TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0         0                1.0  160.0  200.0  180.0  200.0  200.0   
1         1                NaN    NaN    NaN    NaN    NaN    NaN   
2         0                1.0  160.0  120.0  200.0  200.0  200.0   
3         1                1.0   80.0  160.0  120.0  100.0  100.0   
4         1                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0         B  
1           NaN         A  
2         880.0         A  
3         560.0         B  
4           NaN         B  

[5 rows x 29 columns]

In [ ]:
# Renomeando linhas de um dataframe
# Renomeando raça
dados1["RACA"] = dados1["RACA"].replace({0:"nao_declarado", 1:"branca", 2:"preta", 3:"parda", 4:"amarela", 5:"indigena"})

In [30]:
# Renomeando lingua estrangeira
dados1["TP_LINGUA"] = dados1["TP_LINGUA"].replace({0:"Inglês", 1:"Espanhol"})

In [31]:
# Renomeando tipo de escola
dados1["ESCOLA"] = dados1["ESCOLA"].replace({1:"não_respondeu", 2:"pública", 3:"privada", 4:"exterior"})

In [32]:
# Renomeando questão Q25 sobre internet
dados1["INTERNET"] = dados1["INTERNET"].replace({"A":"Não", "B":"Sim"})

In [33]:
dados1.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  não_respondeu             0  ...   
1                 1                1  não_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        pública             0  ...   
4                 1                1  não_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Inglês                1.0  160.0  200.0  180.0  200.0  200.0   
1      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
2    731.8    Inglês                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
4      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
1           NaN       Não  
2         880.0       Não  
3         560.0       Sim  
4           NaN       Sim  

[5 rows x 29 columns]

### Análise Exploratória e Transformação dos Dados

In [ ]:
# Contagem dos valores da coluna IDADE
dados1['IDADE'].value_counts()

IDADE
17    219313
18    147548
19     81236
16     59316
20     53009
       ...  
91         1
4          1
3          1
86         1
2          1
Name: count, Length: 80, dtype: int64

In [ ]:
# Ordenando a coluna IDADE para verificar erros
dados1['IDADE'].value_counts().sort_index()

IDADE
2     1
3     1
4     1
5     2
6     2
     ..
80    2
81    3
82    2
86    1
91    1
Name: count, Length: 80, dtype: int64

Possível ver valores 'estranhos', como idades menores que 12 anos e maiores que 80.

In [37]:
menor_12 = dados1.query('IDADE < 12')['NO_MUNICIPIO_RESIDENCIA'].value_counts()

In [38]:
menor_12

NO_MUNICIPIO_RESIDENCIA
São Paulo                 5
Tambaú                    1
Ribeirão Preto            1
Bálsamo                   1
São Bernardo do Campo     1
Embu das Artes            1
Queiroz                   1
Santana de Parnaíba       1
Vargem Grande Paulista    1
Auriflama                 1
Araçatuba                 1
Campinas                  1
Name: count, dtype: int64

In [39]:
maior_11 = dados1.loc[dados1.IDADE > 11]

In [40]:
maior_11.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  não_respondeu             0  ...   
1                 1                1  não_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        pública             0  ...   
4                 1                1  não_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Inglês                1.0  160.0  200.0  180.0  200.0  200.0   
1      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
2    731.8    Inglês                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
4      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
1           NaN       Não  
2         880.0       Não  
3         560.0       Sim  
4           NaN       Sim  

[5 rows x 29 columns]

In [41]:
maior_11['IDADE'].value_counts().sort_index()

IDADE
12        6
13       29
14      355
15    11583
16    59316
      ...  
80        2
81        3
82        2
86        1
91        1
Name: count, Length: 73, dtype: int64

In [42]:
# Treineiros

maior_11.query('IN_TREINEIRO == 1')['IN_TREINEIRO'].value_counts()

IN_TREINEIRO
1    93988
Name: count, dtype: int64

In [43]:
# Nova tabela de dados - Treineiros
treineiros = maior_11.loc[maior_11.IN_TREINEIRO == 1]

In [44]:
treineiros.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
20      1,90E+11                  Santos     15    F                1   parda   
21      1,90E+11                Panorama     16    F                1   parda   
106     1,90E+11               São Paulo     16    F                1  branca   
109     1,90E+11        Águas de Lindóia     16    M                1  branca   
159     1,90E+11        Águas de Lindóia     15    M                1  branca   

     TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
20                  1                3  não_respondeu             1  ...   
21                  1                3  não_respondeu             1  ...   
106                 1                3  não_respondeu             1  ...   
109                 1                3  não_respondeu             1  ...   
159                 1                3  não_respondeu             1  ...   

     NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
20     573.2    Inglês                1.0  140.0  120.0  120.0  100.0   80.0   
21       NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
106    532.6    Inglês                1.0  120.0  120.0  120.0  100.0   80.0   
109      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
159      NaN    Inglês                NaN    NaN    NaN    NaN    NaN    NaN   

     NOTA_REDACAO  INTERNET  
20          560.0       Sim  
21            NaN       Não  
106         540.0       Sim  
109           NaN       Sim  
159           NaN       Sim  

[5 rows x 29 columns]

In [45]:
# Criando arquivo apenas de Treineiros
treineiros.to_csv('treineiros_enem_2019_sp.csv', encoding='iso-8859-1', index=False)

In [46]:
# Trabalhando com os vestibulandos 
# Separação dos Vestibulandos dos Treineiros

vestibulandos = maior_11.loc[maior_11.IN_TREINEIRO == 0]

In [47]:
vestibulandos.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
1     1,90E+11             Pinhalzinho     19    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
4     1,90E+11    São Joaquim da Barra     37    M                2  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  não_respondeu             0  ...   
1                 1                1  não_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        pública             0  ...   
4                 1                1  não_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Inglês                1.0  160.0  200.0  180.0  200.0  200.0   
1      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   
2    731.8    Inglês                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
4      NaN  Espanhol                NaN    NaN    NaN    NaN    NaN    NaN   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
1           NaN       Não  
2         880.0       Não  
3         560.0       Sim  
4           NaN       Sim  

[5 rows x 29 columns]

In [48]:
vestibulandos.shape

(719768, 29)

 - Analisando as Presenças

In [49]:
# Analisando as presenças por disciplina

vestibulandos['TP_PRESENCA_CN'].value_counts()

TP_PRESENCA_CN
1    500805
0    218693
2       270
Name: count, dtype: int64

In [50]:
vestibulandos['TP_PRESENCA_CH'].value_counts()

TP_PRESENCA_CH
1    533142
0    185969
2       657
Name: count, dtype: int64

In [51]:
vestibulandos['TP_PRESENCA_LC'].value_counts()

TP_PRESENCA_LC
1    533142
0    185969
2       657
Name: count, dtype: int64

In [52]:
vestibulandos['TP_PRESENCA_MT'].value_counts()

TP_PRESENCA_MT
1    500805
0    218693
2       270
Name: count, dtype: int64

RESUMO DA PRESENÇA NAS PROVAS:

Ciências da Natureza: 500805 presentes, 218693 faltaram e 270 eliminados.

Ciências Humanas: 533142 presentes, 185969 faltaram e 657 eliminados.

Linguagens e Códigos: 533142 presentes, 185969 faltaram e 657 eliminados.

Matemática: 500805 presentes, 218693 faltaram e 270 eliminados.

In [53]:
# Situação da Redação do participante
vestibulandos['TP_STATUS_REDACAO'].value_counts().sort_index()

TP_STATUS_REDACAO
1.0    519763
2.0       913
3.0      1417
4.0      5513
6.0      3626
7.0       543
8.0       828
9.0       539
Name: count, dtype: int64

- Valores Faltantes

In [54]:
# Valores Ausentes
vestibulandos['IDADE'].isnull().sum()

np.int64(0)

In [55]:
provas = ['NOTA_REDACAO','NOTA_CN','NOTA_MT','NOTA_LC','NOTA_CH']

In [56]:
vestibulandos[provas].isnull().sum()

NOTA_REDACAO    186626
NOTA_CN         218963
NOTA_MT         218963
NOTA_LC         186626
NOTA_CH         186626
dtype: int64

In [57]:
# Classificando os vestibulandos que compareceram nos dias das provas
vestibulandos_classificados = vestibulandos.loc[vestibulandos.TP_PRESENCA_CH == 1]
vestibulandos_classificados = vestibulandos_classificados.loc[vestibulandos_classificados.TP_PRESENCA_MT == 1]

In [58]:
# Visualizando a nova contagem da nova tabela
vestibulandos_classificados['TP_PRESENCA_MT'].value_counts()

TP_PRESENCA_MT
1    499940
Name: count, dtype: int64

In [59]:
vestibulandos_classificados['TP_PRESENCA_CH'].value_counts()

TP_PRESENCA_CH
1    499940
Name: count, dtype: int64

In [60]:
vestibulandos_classificados['TP_PRESENCA_CN'].value_counts()

TP_PRESENCA_CN
1    499940
Name: count, dtype: int64

In [61]:
vestibulandos_classificados['TP_PRESENCA_LC'].value_counts()

TP_PRESENCA_LC
1    499940
Name: count, dtype: int64

In [62]:
vestibulandos_classificados[provas].isnull().sum()

NOTA_REDACAO    0
NOTA_CN         0
NOTA_MT         0
NOTA_LC         0
NOTA_CH         0
dtype: int64

 - Notas zeros na Redação

In [63]:
vestibulandos_classificados.query('NOTA_REDACAO == 0')['NOTA_REDACAO'].value_counts()

NOTA_REDACAO
0.0    9181
Name: count, dtype: int64

In [64]:
# Verificando separadamente as competências
vestibulandos_classificados.query('COMP1 == 0')['COMP1'].value_counts()

COMP1
0.0    9188
Name: count, dtype: int64

In [66]:
vestibulandos_classificados.query('COMP2 == 0')['COMP2'].value_counts()

COMP2
0.0    9181
Name: count, dtype: int64

In [67]:
vestibulandos_classificados.query('COMP3 == 0')['COMP3'].value_counts()

COMP3
0.0    9197
Name: count, dtype: int64

In [68]:
vestibulandos_classificados.query('COMP4 == 0')['COMP4'].value_counts()

COMP4
0.0    9194
Name: count, dtype: int64

In [69]:
vestibulandos_classificados.query('COMP5 == 0')['COMP5'].value_counts()

COMP5
0.0    73330
Name: count, dtype: int64

In [70]:
vestibulandos_classificados.query('NOTA_CH == 0')['NOTA_CH'].value_counts()

NOTA_CH
0.0    343
Name: count, dtype: int64

In [72]:
vestibulandos_classificados.query('NOTA_CN == 0')['NOTA_CN'].value_counts()

NOTA_CN
0.0    36
Name: count, dtype: int64

In [73]:
vestibulandos_classificados.query('NOTA_LC == 0')['NOTA_LC'].value_counts()

NOTA_LC
0.0    135
Name: count, dtype: int64

In [75]:
vestibulandos_classificados.query('NOTA_MT == 0')['NOTA_MT'].value_counts()

NOTA_MT
0.0    56
Name: count, dtype: int64

NOTAS ZEROS DOS INSCRITOS CONCORRENTES AO VESTIBULAR:

Matemática = 56

Ciências da Natureza = 36

Linguagens e Códigos = 135

Ciências Humanas = 343

Redação = 9181


In [ ]:
# Nova tabela
vestibulandos_classificados.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
5     1,90E+11              Embu-Guaçu     17    F                1   parda   
8     1,90E+11               Guarulhos     23    F                1  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA  IN_TREINEIRO  ...  \
0                 1                1  não_respondeu             0  ...   
2                 1                2        privada             0  ...   
3                 1                2        pública             0  ...   
5                 1                2        pública             0  ...   
8                 1                1  não_respondeu             0  ...   

   NOTA_MT TP_LINGUA  TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  \
0    600.2    Inglês                1.0  160.0  200.0  180.0  200.0  200.0   
2    731.8    Inglês                1.0  160.0  120.0  200.0  200.0  200.0   
3    393.2  Espanhol                1.0   80.0  160.0  120.0  100.0  100.0   
5    419.2    Inglês                1.0  120.0  120.0  120.0  120.0   80.0   
8    796.1    Inglês                1.0  160.0  200.0  180.0  180.0  160.0   

   NOTA_REDACAO  INTERNET  
0         940.0       Sim  
2         880.0       Não  
3         560.0       Sim  
5         560.0       Sim  
8         880.0       Sim  

[5 rows x 29 columns]

In [77]:
vestibulandos_classificados = vestibulandos_classificados.drop(columns=['IN_TREINEIRO'], )

In [82]:
# Dataframe tratado!
vestibulandos_classificados.head()

NU_INSCRICAO NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0     1,90E+11            Santa Isabel     22    M                1   parda   
2     1,90E+11               São Paulo     17    F                1   parda   
3     1,90E+11   São Bernardo do Campo     19    M                1   parda   
5     1,90E+11              Embu-Guaçu     17    F                1   parda   
8     1,90E+11               Guarulhos     23    F                1  branca   

   TP_NACIONALIDADE  TP_ST_CONCLUSAO         ESCOLA   CO_ESCOLA  \
0                 1                1  não_respondeu         NaN   
2                 1                2        privada  15567761.0   
3                 1                2        pública  35904958.0   
5                 1                2        pública         NaN   
8                 1                1  não_respondeu         NaN   

        NO_MUNICIPIO_ESC  TP_PRESENCA_CN  TP_PRESENCA_CH  TP_PRESENCA_LC  \
0                    NaN               1               1               1   
2                  Belém               1               1               1   
3  São Bernardo do Campo               1               1               1   
5                    NaN               1               1               1   
8                    NaN               1               1               1   

   TP_PRESENCA_MT  NOTA_CN  NOTA_CH  NOTA_LC  NOTA_MT TP_LINGUA  \
0               1    564.6    585.8    592.9    600.2    Inglês   
2               1    576.2    641.1    634.3    731.8    Inglês   
3               1    402.2    424.8    509.6    393.2  Espanhol   
5               1    399.6    426.8    487.2    419.2    Inglês   
8               1    625.7    646.5    629.3    796.1    Inglês   

   TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO INTERNET  
0                1.0  160.0  200.0  180.0  200.0  200.0         940.0      Sim  
2                1.0  160.0  120.0  200.0  200.0  200.0         880.0      Não  
3                1.0   80.0  160.0  120.0  100.0  100.0         560.0      Sim  
5                1.0  120.0  120.0  120.0  120.0   80.0         560.0      Sim  
8                1.0  160.0  200.0  180.0  180.0  160.0         880.0      Sim

In [80]:
pd.options.display.max_columns = 44

In [81]:
vestibulandos_classificados.shape

(499940, 28)

### Salvando e Exportando o Dataframe Tratado

In [83]:
vestibulandos_classificados.to_csv('enem_2019_tratado.csv', encoding='iso-8859-1', index=False)